packages/spacecat-shared-data-access/src/models/api-key.js

```diff
@@ -82,12 +82,14 @@ export const createApiKey = (data) => {
       throw new Error(`Scope name is not part of the pre-defined scopes: ${scope.name}`);
     }
 
-    if (hasText(scope.domains) && !Array.isArray(scope.domains)) {
-      throw new Error(`Scope domains should be an array: ${scope.domains}`);
-    }
-    for (const domain of scope.domains) {
-      if (!isValidUrl(domain)) {
-        throw new Error(`Invalid domain: ${domain}`);
+    if (scope.domains) {
```

Comment:
Looks like this changes the behaviour (_depending on `spacecat-shared-utils/hasText` I suppose_).  If `scope.domains` is `null/undefined` the call will not throw an error like it did before (right?).
Is this intentional?

Reply:
This code was added by us recently for this feature - not in use yet by others. There was a logic issue that this PR is correcting: `scope.domains` is optional, and in the case where it’s not set the for loop was throwing an error (“scope.domains is not Iterable”). It is not a text property so `hasText` was not the correct check, either.